source/simplex/fe_lib.cc

```diff
@@ -69,60 +69,52 @@ namespace Simplex
         {
           // We don't really have dim = 1 support for simplex elements yet, but
           // its convenient for populating the face array
+          Assert(degree <= 2, ExcNotImplemented());
           if (degree >= 1)
             {
               unit_points.emplace_back(0.0);
               unit_points.emplace_back(1.0);
-            }
-          if (degree == 2)
-            {
-              unit_points.emplace_back(0.5);
-            }
-          if (degree > 2)
-            {
-              Assert(false, ExcNotImplemented());
+
+              if (degree == 2)
+                unit_points.emplace_back(0.5);
             }
         }
       else if (dim == 2)
         {
+          Assert(degree <= 2, ExcNotImplemented());
```

Comment:
The new condition is different than the previous one. This looks correct to me but I just want to make sure that it was a deliberate change and not simply a copy/paste.

Reply:
I think the code before was wrong. It had
```
          if (degree >= 1)
            {
              unit_points.emplace_back(0.0, 0.0);
              unit_points.emplace_back(1.0, 0.0);
              unit_points.emplace_back(0.0, 1.0);
            }
          if (degree == 2)
            {
              unit_points.emplace_back(0.5, 0.0);
              unit_points.emplace_back(0.5, 0.5);
              unit_points.emplace_back(0.0, 0.5);
            }
          if (degree > 3)
            {
              Assert(false, ExcNotImplemented());
            }
```
This doesn't cover the case `degree==3` at all, but I think that was a mistake. `degree==3` should trigger the exception, which the changed assertion actually does.

To be fair, I had not seen this issue when writing the patch -- good eyes, @Rombur. It's a bit amusing that the existing code treated the exceptional case in three different ways:
```
      if (dim == 1)
        {
          // We don't really have dim = 1 support for simplex elements yet, but
          // its convenient for populating the face array
          if (degree >= 1)
            {
              unit_points.emplace_back(0.0);
              unit_points.emplace_back(1.0);
            }
          if (degree == 2)
            {
              unit_points.emplace_back(0.5);
            }
          if (degree > 2)
            {
              Assert(false, ExcNotImplemented());
            }
        }
      else if (dim == 2)
        {
          if (degree >= 1)
            {
              unit_points.emplace_back(0.0, 0.0);
              unit_points.emplace_back(1.0, 0.0);
              unit_points.emplace_back(0.0, 1.0);
            }
          if (degree == 2)
            {
              unit_points.emplace_back(0.5, 0.0);
              unit_points.emplace_back(0.5, 0.5);
              unit_points.emplace_back(0.0, 0.5);
            }
          if (degree > 3)
            {
              Assert(false, ExcNotImplemented());
            }
        }
      else if (dim == 3)
        {
          if (degree >= 1)
            {
              unit_points.emplace_back(0.0, 0.0, 0.0);
              unit_points.emplace_back(1.0, 0.0, 0.0);
              unit_points.emplace_back(0.0, 1.0, 0.0);
              unit_points.emplace_back(0.0, 0.0, 1.0);
            }
          if (degree == 2)
            {
              unit_points.emplace_back(0.5, 0.0, 0.0);
              unit_points.emplace_back(0.5, 0.5, 0.0);
              unit_points.emplace_back(0.0, 0.5, 0.0);
              unit_points.emplace_back(0.0, 0.0, 0.5);
              unit_points.emplace_back(0.5, 0.0, 0.5);
              unit_points.emplace_back(0.0, 0.5, 0.5);
            }
          if (degree == 3)
            {
              Assert(false, ExcNotImplemented());
            }
```
I believe that two of the three ways were wrong, and are now correct :-)

Reply:
I think before (yesterday or so) it was `else {Assert(false, ExcNotImplemented());}` in all cases ;) But back then, the vertex points were duplicated... This way it is much better!